modules/core/src/lib/attribute.js

```diff
@@ -38,9 +59,12 @@ export default class Attribute extends BaseAttribute {
 
     this.shaderAttributes = {};
     this.hasShaderAttributes = false;
+    this.doublePrecision = doublePrecision;
 
-    if (opts.shaderAttributes) {
-      const shaderAttributes = opts.shaderAttributes;
+    if (opts.shaderAttributes || doublePrecision) {
```

Comment:
Nit: Not new for this PR, but I notice that an `Attribute` has a map of `Attribute`s. Looks a little hacky, wonder if we would benefit from separating into `LogicalAttribute` and `GPUAttribute` or similar once the dust settles.